synapse/event_auth.py

```diff
@@ -760,4 +855,13 @@ def auth_types_for_event(event: Union[EventBase, EventBuilder]) -> Set[Tuple[str
                 )
                 auth_types.add(key)
 
+        # TODO Should this be limited to only MSC3083 rooms.
```

Comment:
I'm thinking yes, which means we'll need to pipe the room version in here.

Reply:
Yeah, I think it makes sense for `auth_types_for_event` to be room version specific.